### PR TITLE
Prevent certain cutoffs from being selected based on round Format

### PR DESCRIPTION
**Pull request by coder13**
fixes part of #1916.
Also bundled in a change with the text in the cutoff modal saying "Round Format" changing it to "Cutoff Format" which makes more sense, should probably have been a separate change however.

### CODE REVIEW

**Comment by viroulep** [comment]
@coder13, sorry I totally missed this, but with the new 9b1 only Bo2 is possible as a combined format for Ao5 rounds!

Could you please send a PR to reflect the changes in the Regulations?

**Comment by coder13** [comment]
Sure. My solution just works really simply to prevent setting cutoffs for more solves than you actually have, I can further restrict it by round type.

Do we also want to prevent setting cutoffs for Best of X / mean events- round formats that can't be combined?

**Comment by viroulep** [comment]
It would be awesome if we could have the following behavior:
 is:open 
 - if the round format is Ao5, then only Bo2 is allowed for cutoff.
 - else if the round format is Mo3, then Bo1 and Bo2 are allowed.
 - else if the preferred format for the event is not Ao5, the cutoff must be < X in the BoX format.
 - else no cutoff.

I think this would support all formats in 9b!

File: WcaOnRails/app/javascript/edit-events/modals/Cutoff.jsx [hunk 1]
```diff
@@ -68,10 +69,10 @@ export default {
...
               >
                 <option value={0}>No cutoff</option>
-                <option disabled="disabled">────────</option>
-                <option value={1}>Best of 1</option>
-                <option value={2}>Best of 2</option>
-                <option value={3}>Best of 3</option>
+                {solveCount > 1 ? <option disabled="disabled">────────</option> : null}
+                {solveCount > 1 && wcifRound.format !== 'a'  ? <option value={1}>Best of 1</option> : null}
+                {solveCount > 2 ? <option value={2}>Best of 2</option> : null}
+                {solveCount > 3 && wcifRound.format !== 'a' ? <option value={3}>Best of 3</option> : null}
```
[the remark this answers is not in the document]
Comment (coder13) [reply WcaOnRails/app/javascript/edit-events/modals/Cutoff.jsx]: Good suggestion, I'll make the change.

**Comment by jfly** [comment]
LGTM! I'm liking the refactor :)

**Comment by viroulep** [comment]
Thanks @coder13!

**Pull request merged**